src/main/java/com/stripe/model/TypeDataDeserializer.java

```diff
@@ -0,0 +1,53 @@
+package com.stripe.model;
+
+import com.google.gson.*;
+
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.HashMap;
+
+public class TypeDataDeserializer<T extends HasTypeData> implements JsonDeserializer<T> {
```

Comment:
Would you mind giving this a slightly more specific name and adding a doc block explaining roughly what it does to the class? I think that "type data deserializer" is just a little too generic given that this is all related to sources.

Sorry definitely a nit given that you were not the one to originally cause these issues :$